README.md

```diff
@@ -1,40 +1,81 @@
 
 ## Getting Started
 
-### Prerequisites
+CodeShovel can be used in three ways:
+
+* ***Web Service UI***: We have built a web based user interface that you can use to interactively navigate the history of a method of interest. We host a public copy of the [web interface](https://se.cs.ubc.ca/codeshovel/index.html) if you just want to use CodeShovel without installing anything, but this repository also has [instructions for self-hosting](#web-service-ui) the web service on your own computer using Docker.
+
+* ***Web Service REST:*** If you would prefer to programmatically interact with the history of your methods, you can also call the CodeShovel web service using standard rest commands. You can direct these against our hosted version of the web service, or against your own self-hosted copy of the web service. The [REST interface instructions](#web-service-rest) are below.
+
+* ***Command Line:*** Finally, if you would prefer to interact with CodeShovel on the command line directly without using the rest interface, you can call the CodeShovel `jar` directly. [Command line instructions](#command-line) are included below.
+
+<a name="web-service-ui"></a>
+### Web Service UI 
+
+***Public UI:*** The web service UI enables easy interactive exploration of a method history. The quickest way to use this is through our hosted version available at [https://se.cs.ubc.ca/codeshovel/index.html](https://se.cs.ubc.ca/codeshovel/index.html). Through this interface you can explore histories of some sample methods (these are not cached: they are dynamically computed as the underlying repositories are updated), or by providing a link to a public repository of your choosing.
+
+***Self-Hosted UI:*** You can also stand up a copy of the web interface on your own infrastructure. To do this, follow these steps:
+
+1. Clone the repostitory: `git clone git@github.com:ataraxie/codeshovel.git`
+2. Edit the `.env` file to point to the right directories for caches.
+3. Start the web service: `docker-compose build && docker-compose up`
+4. Open the web service in your browser: `https://localhost:8000`
 
-* Java 8 or higher must be installed (`java -version` must reveal 1.8 or higher). You can find Java on the [Oracle page](https://www.oracle.com/technetwork/java/javase/downloads/index.html).
-* If you want to build CodeShovel yourself: Maven 3.5.x or higher
+<a name="web-service-rest"></a>
+### Web Service REST 
 
-### Running the tool from command line
+As with the web service UI, you can either use our public web service or self-host your own (see the instructions [#web-service-ui](above). If you are using the self-hosted web service, change `https://se.cs.ubc.ca/codeshovel` to point to your own service.
+
+You can then interact with the host using the following REST endpoints. Examples are provided using `curl` syntax for ease of testing, just adapt the values as needed.
+
+* To get the history for a method:
+
+```
+curl TBD
```

Comment:
@braxtonhall is there any chance you could give us a curl command for each of the three endpoints and add them here (or slack them at me)?